README.md

```diff
@@ -1 +1,134 @@
-# stencil-progress-bar
+# Stencil Progress Bar
+
+A progress bar that comes in both standard and radial flavours. 
+
+[Demo Coming Soon](#)
+
+
+
+## Requirements
+
+- Sass
+- AMD loader
+- Dust templating
+
+
+
+## Installation
+
+Install the Stencil Progress Bar using the AdaptiveJS generator:
+```shell
+yo adaptivejs:component --install stencil-progress-bar
+```
+You can also install interactively by running `yo adaptivejs:component` and choosing the install option.
+
+
+
+## Usage
+
+Stencil components installed with the AdaptiveJS generator require no setup. Just load the component’s template in the desired view and use the dust helper to render it. For details, see [using Components in AdaptiveJS](https://github.com/mobify/devcenter-assets/blob/master/mobifydevcenter/v2.0/documentation/Components/Use%20a%20Component.md).
+
+
+
+## API
+
+### Template
+The template accepts the following parameters.
+
+Name | Type | Description
+-----|------|------------
+value | number within [0, 1] | Sets the initial value for the progress bar
+label | string | Sets the initial label for the progress bar
+state | one of ['inprogress', 'success', 'failure'] | Sets the initial state for the progress bar
+isRadial | boolean | If `true`, uses the radial progress bar instead of the linear progress bar
+
+This component has no template bodies.
+
+
+### UI Script
+
+First, require the Progress Bar UI script. Then call `ProgressBar.init`.
+
+```js
+require([
+    // ...
+    'progress-bar-ui'
+], function(
+    // ...
+    ProgressBar
+) {
+    ProgressBar.init($el);
+});
+```
+
+`ProgressBar.init` takes one parameter: `$el`.
+
+#### $el
+
+`$el` is the progress bar element. The component's instance is accessible through this DOM node. For instance, the following calls the `setProgress` method on an instance of the Progress Bar component:
+
+```js
+$('.c-progress-bar').data('progressbar').setProgress(0.5);
```

Comment:
Right now, the way components are architected, the recommended way to use them is via their dust helper. So a button component is added to the page with this Dust: `{@c-button}My button label{/c-button}`. If the button were to ship with a UI script, it would automatically be invoked by Adaptive against that DOM node. You could invoke it yourself, if you wanted to pass custom options, like this:

```javascript
var $progress = $('#my-progress-bar');
var customOptions = {/* some options */};
window.Adaptive.Component.init($progress, options);
```

So the framework explicitly discourages invoking init yourself, and mixing and matching the behaviour of one component with the DOM of another. The boilerplate init implementation I have in the Select component (which I’d like to evolve into the standard) presently looks like this (so you can’t double-instantiate):

```
return {
    'init': function($el, options) {
        // If already initialized, return the instance; otherwise, create it
        // and expose it through `$('.c-select').data('component')`.
        return $el.data('component') || $el.data('component', new Select($el, options));
    }
};
```